packages/cli/src/ui/components/HookStatusDisplay.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type React from 'react';
+import { Text } from 'ink';
+import { theme } from '../semantic-colors.js';
+import { useTerminalSize } from '../hooks/useTerminalSize.js';
+import { type ActiveHook } from '../types.js';
+
+interface HookStatusDisplayProps {
+  activeHooks: ActiveHook[];
+}
+
+export const HookStatusDisplay: React.FC<HookStatusDisplayProps> = ({
+  activeHooks,
+}) => {
+  const { columns: terminalWidth } = useTerminalSize();
+
+  const hooksText = (() => {
+    if (activeHooks.length === 0) {
+      return '';
+    }
+
+    const label = activeHooks.length > 1 ? 'Executing Hooks' : 'Executing Hook';
+    let currentChars = label.length + 2; // label + ": " (2)
+    const displayedHooks = [];
+    let hiddenCount = 0;
+
+    // Use a larger portion of the screen since we are overwriting the summary
+    const maxChars = Math.floor(terminalWidth * 0.8);
```

Comment:
you are truncating to one line. you should be able to use the built in Ink support to truncate to a line that will add a nicer "..." char as well as be correct at text measurement. alternately if you want to customize create a new UI component that does the correct measurement (can use text-buffer as example of where we do correct text measurement and wrapping today.